client/me/help/help-contact/index.jsx

```diff
@@ -412,18 +421,118 @@ const HelpContact = React.createClass( {
+	getSupportVariation: function() {
+		const { olark } = this.state;
+		const { ticketSupportEligible } = this.props;
+
+		if ( this.shouldUseHappychat() ) {
+			return SUPPORT_HAPPYCHAT;
+		}
+
+		if ( olark.isUserEligible && olark.isOperatorAvailable ) {
+			return SUPPORT_LIVECHAT;
+		}
+
+		if ( olark.details.isConversing || ticketSupportEligible ) {
+			return SUPPORT_TICKET;
+		}
+
+		return SUPPORT_FORUM;
+	},
+
+	getContactFormPropsVariation: function( variationSlug ) {
+		const { isSubmitting } = this.state;
+		const { translate } = this.props;
+		const moreThanOneSite = sites.get().length > 1;
+
+		switch ( variationSlug ) {
+			case SUPPORT_HAPPYCHAT:
+				const isDev = ( ( config( 'env' ) === 'development' ) || ( config( 'env_id' ) === 'stage' ) );
+				return {
+					onSubmit: this.startHappychat,
+					buttonLabel: isDev ? 'Happychat' : translate( 'Chat with us' ),
+					showSubjectField: false,
+					showHowCanWeHelpField: true,
+					showHowYouFeelField: true,
+					showSiteField: moreThanOneSite,
+				};
+
+			case SUPPORT_LIVECHAT:
+				return {
+					onSubmit: this.startChat,
+					buttonLabel: translate( 'Chat with us' ),
+					showSubjectField: false,
+					showHowCanWeHelpField: true,
+					showHowYouFeelField: true,
+					showSiteField: moreThanOneSite,
+				};
+
+			case SUPPORT_TICKET:
+				return {
+					onSubmit: this.submitKayakoTicket,
+					buttonLabel: isSubmitting ? translate( 'Sending email' ) : translate( 'Email us' ),
+					showSubjectField: true,
+					showHowCanWeHelpField: true,
+					showHowYouFeelField: true,
+					showSiteField: moreThanOneSite,
+				};
+
+			default:
+				return {
+					onSubmit: this.submitSupportForumsTopic,
+					buttonLabel: isSubmitting ? translate( 'Asking in the forums' ) : translate( 'Ask in the forums' ),
+					formDescription: translate(
+						'Post a new question in our {{strong}}public forums{{/strong}}, ' +
+						'where it may be answered by helpful community members, ' +
+						'by submitting the form below. ' +
+						'{{strong}}Please do not{{/strong}} provide financial or ' +
+						'contact information when submitting this form.',
+						{
+							components: {
+								strong: <strong />
+							}
+						} ),
+					showSubjectField: true,
+					showHowCanWeHelpField: false,
+					showHowYouFeelField: false,
+					showSiteField: false,
+				};
+		}
+	},
+
+	getContactFormCommonProps: function() {
+		const { olark, isSubmitting } = this.state;
+		const showHelpLanguagePrompt = ( olark.locale !== i18n.getLocaleSlug() );
+
+		return {
+			disabled: isSubmitting,
+			showHelpLanguagePrompt: showHelpLanguagePrompt,
+			valueLink: { value: savedContactForm, requestChange: ( contactForm ) => savedContactForm = contactForm }
+		};
+	},
+
+	shouldShowTicketRequestErrorNotice: function( variationSlug ) {
+		const { ticketSupportRequestError } = this.props;
+
+		return SUPPORT_HAPPYCHAT !== variationSlug && SUPPORT_LIVECHAT !== variationSlug && null !== ticketSupportRequestError;
```

Comment:
For `null` checks it might be better to use `!=` instead of `!==`, since `null !== undefined` returns true. Or  just check for truthiness with `!!ticketSupportRequestError`.